src/core/lib/security/util/b64.h

```diff
@@ -41,6 +41,17 @@
 char *grpc_base64_encode(const void *data, size_t data_size, int url_safe,
                          int multiline);
 
+/* estimate the upper bound on size of base64 encoded data. The actual size
+ * is guaranteed to be less than or equal to the size returned here. */
+size_t grpc_base64_estimate_encoded_size(size_t data_size, int url_safe,
```

Comment:
I'd like to do this in a different PR so we keep the focus of this PR In just adding functionality.